### DISCUSSION TITLE
Spotify API 사용을 변경하려고 합니다.

### DISCUSSION THREAD

**Discussion by hyunbin78** [prgrms-fe-devcourse/NFE1_2_2_MUX#53] [D. 아이디어&레퍼런스]
Spotify API를 사용하려면 로그인 해야되는 방식이 앨범을 가져오는데는 문제 없지만 플레이를 하기에는 문제가 생겨서 앨범을 가져오는것은 문제 없지만 플레이가 안되고 있는 상황이라 Youtube-Music API를 사용할 예정입니다. 또한 앨범 가져오기를 누르면 검색을 할 수 있게 기능 추가 예정입니다.

- 해결방안 
1. 앨범 정보를 가져오기 위해서는 검색을 해야할 것 같아서 검색 부분도 모달 내에 추가하였습니다.
2. Youtube-Music API를 사용하여 앨범 정보를 가져오게끔 하였습니다.
3. react-youtube 라이브러리 설치: react-youtube는 유튜브 IFrame API를 쉽게 사용할 수 있도록 해주는 React 컴포넌트를 사용해서 재생을 앨범에 대한 곡 재생을 불러올 수 있게 할 예정입니다. 또한  API(특히 YouTube-music API)로 데이터를 요청할 때, 브라우저의 CORS 정책에 의해 오류가 발생하므로 package-json에서  프록시를 설정하여 로컬 서버에서 API 요청을 우회하는 방법을 사용합니다.
아래의 코드를 package-json에 추가할 예정입니다. 
 "proxy": "https://youtube-music-api3.p.rapidapi.com"

위 해결 방안으로 구현해 놓았는데 PR은 아직 올리지 않은 상태입니다. 수정 가능한 부분이니 의견 주시면 감사하겠습니다!

**Comment by ryan-g00** [comment]
결국 스포티파이와의 싸움에서 지셨군요 현빈님... ㅠㅠ
